File: zbg-java-sdk-api/src/main/java/com/dd/api/api/MarketApi.java
package com.dd.api.api;

import com.dd.api.entity.commom.result.TradeHistory;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Headers;
import retrofit2.http.Path;
import retrofit2.http.Query;

import java.util.List;
import java.util.Map;

/**
 * 行情数据
 *
 * @author zhangzp
 */
public interface MarketApi {
    /**
     * 返回历史K线数据。（蜡烛图）
     *
     * @param symbol   交易对名称
     * @param type     K线类型,支持1M，5M，15M，30M，1H，1D，1W
     *                 七种类型，分别代表1-30分钟，1小时，1日，1周
     * @param dataSize 返回 K 线数据条数,[1,100]
     */
    @Headers({"url_name:kline"})
    @GET("https://kline.zbg.com/api/data/v1/klines")
    Call<List<List<String>>> getKlines(@Query("marketName") String symbol, @Query("type") String type, @Query("dataSize") Integer dataSize);

    /**
     * 获取ticker信息同时提供最近24小时的交易聚合信息。
     *
     * @param symbol 交易对名称
     */
    @Headers({"url_name:kline"})
    @GET("https://kline.zbg.com/api/data/v1/ticker")
    Call<List<String>> getTicker(@Query("marketName") String symbol);

    /**
     * 获得所有交易对的 tickers，数据取值时间区间为24小时滚动。
     *
     * @param isUseMarketName 是否返回交易对名称
     */
    @Headers({"url_name:kline"})
    @GET("https://kline.zbg.com/api/data/v1/tickers")
    Call<Map<String, List<String>>> getTickers(@Query("isUseMarketName") boolean isUseMarketName);

    /**
     * 返回历史K线数据。（蜡烛图）
     *
     * @param symbol   交易对名称
     * @param dataSize 返回数据条数，默认80，最多为1000
     */
    @Headers({"url_name:kline"})
    @GET("https://kline.zbg.com/api/data/v1/trades")
    Call<List<List<String>>> getTrades(@Query("marketName") String symbol, @Query("dataSize") Integer dataSize);

    /**
     * 返回指定交易对的当前市场深度数据。
     *
     * @param symbol   交易对名称
     * @param dataSize 档位数，表示买卖各5档，最大为100
     */
    @Headers({"url_name:kline"})
    @GET("https://kline.zbg.com/api/data/v1/entrusts")
    Call<Map<String, Object>> getPriceDepth(@Query("marketName") String symbol, @Query("dataSize") Integer dataSize);

    /**
     * 查询交易对的近80历史成交记录
     *
     * @param symbol 交易对名称
     */
    @GET("/exchange/api/v1/common/trade-history/{symbol}")
    Call<List<TradeHistory>> getHistoryTrades(@Path("symbol") String symbol);

    /**
     * 返回从[trade-id]往后的最多1000历史成交记录：
     *
     * @param symbol  交易对名称
     * @param tradeId 限制的成交记录ID
     */
    @GET("/exchange/api/v1/common/trade-history/{symbol}/{trade-id}")
    Call<List<TradeHistory>> getHistoryTrades(@Path("symbol") String symbol, @Path("trade-id") String tradeId);
}

File: zbg-python-sdk-api/zbg/market_api.py
"""
Market Data Api
"""
from typing import List

from zbg import ApiClient
from zbg.model.market import *
from zbg.model.constant import *


class MarketApi(ApiClient):

    def __init__(self, market_api_host=None):
        describe = {
            'apis': {
                'public': {
                    'get': {
                        'klines': '/api/data/v1/klines',
                        'ticker': '/api/data/v1/ticker',
                        'tickers': '/api/data/v1/tickers',
                        'price_depth': '/api/data/v1/entrusts',
                        'trades': '/api/data/v1/trades',
                        'historical_trades': '/exchange/api/v1/common/trade-history/{symbol}',
                        'historical_trades_by_id': '/exchange/api/v1/common/trade-history/{symbol}/{trade-id}',
                    },
                },
            }
        }

        if market_api_host:
            describe['urls'] = {
                'market_api': market_api_host,
            }

        super().__init__(config=describe)

    CandleStickList = List[CandleStick]

    def get_candlesticks(self, symbol: str, interval=CandlestickInterval.MIN15, data_size=100) -> CandleStickList:
        self.check_symbol(symbol)

        params = {
            'marketName': symbol,
            'type': interval.value,
            'dataSize': data_size,

        }
        data_array = self.public_get_klines(params)

        return [CandleStick.json_parse(e) for e in data_array]

    def get_ticker(self, symbol: str) -> Ticker:
        self.check_symbol(symbol)
        params = {
            'marketName': symbol,
        }
        data_array = self.public_get_ticker(params)

        ticker = Ticker.json_parse(data_array)
        ticker.symbol = symbol

        return ticker

    Tickers = List[Ticker]

    def get_tickers(self) -> Tickers:
        self.load_markets()
        data_array = self.public_get_tickers()

        tickers = list()
        for item in data_array:
            ticker = Ticker.json_parse(item)
            if ticker.symbol_id in self.markets_by_id:
                ticker.symbol = self.markets_by_id[ticker.symbol_id]['symbol']
                tickers.append(ticker)

        return tickers

    def get_price_depth(self, symbol: str, size=50) -> PriceDepth:
        self.check_symbol(symbol)
        params = {
            'marketName': symbol,
            'dataSize': size
        }

        data_object = self.public_get_price_depth(params)

        return PriceDepth.json_parse(data_object)

    TradeList = List[Trade]

    def get_trades(self, symbol: str, size=80) -> TradeList:
        self.check_symbol(symbol)
        params = {
            'marketName': symbol,
            'dataSize': size
        }

        data_array = self.public_get_trades(params)

        return [Trade.json_parse(data_object) for data_object in data_array]

    HistoricalTradeList = List[HistoricalTrade]

    def get_historical_trades(self, symbol: str, trade_id=None) -> HistoricalTradeList:
        """
        Return The latest 80 transaction records.

        :param symbol: The name of the market.
        :param trade_id: The maximum 1000 historical transaction records from [trade-id] onwards.
        :return: The historical trade records.
        """

        self.check_symbol(symbol)
        params = {
            'symbol': symbol,
            'trade-id': trade_id
        }

        data_array = self.public_get_historical_trades_by_id(params) if trade_id else self.public_get_historical_trades(params)

        return [HistoricalTrade(**data_object) for data_object in data_array]
